src/NorthwindWebsite/Core/ApplicationSettings/AppSettings.cs

```diff
@@ -31,7 +35,20 @@ public class AppSettings : IValidatable
             SerilogConfiguration = configuration.GetSection(nameof(Serilog)).Get<SerilogConfig>(),
             FileUploadOptions = configuration.GetSection(nameof(FileUploadOptions)).Get<FileUploadOptions>(),
             CachingConfigs = configuration.GetSection(nameof(CachingConfigs)).Get<CachingConfigs>(),
-            EmailSenderConfigs = configuration.GetSection(nameof(EmailSenderConfigs)).Get<EmailSenderConfigs>()
+            EmailSenderConfigs = configuration.GetSection(nameof(EmailSenderConfigs)).Get<EmailSenderConfigs>(),
+            MicrosoftAccountConfig = new MicrosoftAccountConfig
+            {
+                ClientId = configuration["Authentication:Microsoft:ClientId"],
+                ClientSecret = configuration["Authentication:Microsoft:ClientSecret"]
+            },
+            AzureAdConfigs = new AzureAdConfigs
```

Comment:
Because it gets values from different sources: insignificant ones - from appsettings, and sensitive - from secrets. As I perceive it, it should work something like that.

Reply:
Understood, in thin case, let's use `nameof(AzureAdConfigs):nameof(AzureAdConfigs.Instance)` approach for building your app settings names, also, you can move `configuration["Authentication:Microsoft:ClientId"]` to separate variable to easily reuse it for `MicrosoftAccountConfig` and `AzureAdConfigs`;